### DISCUSSION TITLE
Regarding the issue of OBS stuttering when enabling adaptive streaming to AMS

### DISCUSSION THREAD

**Discussion by sunpeng222** [ant-media/Ant-Media-Server#5811] [Freelancing]
I am pushing an H264 RTMP stream from OBS to AMS and have enabled adaptive streaming. However, when playing the stream, there is a significant delay in the video, and the message "Broadcasting 0.98x" appears in the playlist. But when I push the stream through a browser with adaptive streaming enabled, it is smooth. Can you please tell me what could be causing this issue?

**Comment by tremzo** [comment]
@sunpeng222 can you make sure you have setup your OBS as recommended here https://antmedia.io/docs/guides/publish-live-stream/rtmp/publish-with-obs/#tune-for-ultra-low-latency-streaming and try again.

**Reply by sunpeng222** [reply]
I have configured everything according to the tutorial except for the audio, but the adaptive streaming still lags.  I have tried both configurations for the audio track.  Is there any other possibility?
![img_v3_0258_c8176e8c-110e-4a9a-a922-79400747299g](https://github.com/ant-media/Ant-Media-Server/assets/27934653/c0ab4e13-1190-4b0e-adf3-7964ccc2c141)
Broadcasting0.99x
![img_v3_0258_960a1da3-92cb-453b-9e78-09149864149g](https://github.com/ant-media/Ant-Media-Server/assets/27934653/e523b4ff-2e65-4557-9ea2-dfc250749a84)
Broadcasting1.00x
As long as you turn off Adaptive Streaming, it will become very smooth.What is the meaning of Broadcasting?It is constantly changing and does not seem to have much to do with smoothness.

**Reply by yashtandon113** [reply]
@sunpeng222 
Let me add some points here.

1. As I see, you are using the medium CPU preset, so it is supposed to use more CPU resources on the system. Please make sure that the CPU is not getting overloaded.
2. What is the resolution and FPS of the stream that you are publishing, and could you please share your ABR streaming settings as well on the server side?

**Reply by sunpeng222** [reply]
Even if I adjust it to very fast, it won't work
![img_v3_0258_2faadd91-f054-4ce4-9653-dc6af32f797g](https://github.com/ant-media/Ant-Media-Server/assets/27934653/0b9fe15f-704a-4e51-8350-fa79da1c50cc)
![img_v3_0258_9e75601c-33c9-4fec-94fc-475da8a7c41g](https://github.com/ant-media/Ant-Media-Server/assets/27934653/fa6cac7b-1dfe-4c6c-942c-311e5c4df782)
![img_v3_0258_5d85cb06-5184-4ce5-b1ef-18eeb7e004dg](https://github.com/ant-media/Ant-Media-Server/assets/27934653/f15c0bae-c2ac-45a9-bcd8-cf0cc11480d8)

**Reply by yashtandon113** [reply]
OK, thanks for sharing the further details.

Could you also share the screenshot from the OBS Video tab so we can see the input resolution and FPS? Basically, if you are sending a 1080p stream, then enabling 1080p on the server side ABR is fine, but if you are sending a 720p stream, then please use the same resolution on the server side as well.
Because forcibly upscaling the resolution also creates this kind of problem, make sure that the FPS value is also the same for better results.

**Reply by sunpeng222** [reply]
Thank you for adjusting the resolution of OBS and AMS to be consistent, which has resulted in significant improvement and smoother operation
![SK2Fv5EU5N](https://github.com/ant-media/Ant-Media-Server/assets/27934653/1122c38a-fcdd-4f2c-9a24-604ee179d014)
![gae5G160Lz](https://github.com/ant-media/Ant-Media-Server/assets/27934653/c94997f7-9073-4990-941c-49b54d6270d8)
![lWH35tqwQ1](https://github.com/ant-media/Ant-Media-Server/assets/27934653/39a837c6-117c-48d8-b14b-8b6b18f494d6)

**Reply by sunpeng222** [reply]
However, if it is set to 1080, it will still be too laggy
720 No problem
![O7ZUio7pze](https://github.com/ant-media/Ant-Media-Server/assets/27934653/75604eee-4b80-448a-980c-0b85a454c834)
![img_v3_0258_e8e8d409-c19e-4a3e-b4f4-4a3371bf07cg](https://github.com/ant-media/Ant-Media-Server/assets/27934653/b7222660-5427-488b-9bfb-0f205f9619f8)
![img_v3_0258_fbb282ab-b364-481a-916a-eaa06d1ccd5g](https://github.com/ant-media/Ant-Media-Server/assets/27934653/40e728ed-2182-4a8e-841f-a5db3b4d1861)

**Reply by yashtandon113** [reply]
Ok, so as I see that you are adding any external source to your OBS, it would be good if you used the custom FPS there as well.

When you are enabling 1080p in a video resolution, then please make sure that you are not **rescaling output** back to 720p in the streaming section, as shown in the below screenshot that you shared, because in that way, it will again send the 720p stream, and on the server, 1080p is enabled.

![image](https://github.com/ant-media/Ant-Media-Server/assets/82374739/9dbeba17-64fc-4369-83ef-d995b2968fdc)

**Reply by sunpeng222** [reply]
Yes, this is my current configuration, but it is still too laggy
![GFiCGleMfi](https://github.com/ant-media/Ant-Media-Server/assets/27934653/93816f24-a1be-4678-839f-88f434018c2b)
![image](https://github.com/ant-media/Ant-Media-Server/assets/27934653/1360b22e-5578-4f7a-a319-4b445c22afa0)
![image](https://github.com/ant-media/Ant-Media-Server/assets/27934653/bd4d49f1-dc05-4ad2-a185-c467169ede00)
![image](https://github.com/ant-media/Ant-Media-Server/assets/27934653/76595af8-fa43-4d7a-9e14-1c15e6606f81)

**Reply by yashtandon113** [reply]
Could you please make the custom FPS to 30 on your source in OBS instead of **Match Output FPS**? Also, please set the bit rate to 2000 kbps in the OBS streaming settings as well.

If possible, please try with 25 FPS as well.

I hope it helps.

**Reply by sunpeng222** [reply]
I tried to set the params you mentioned, but the video is still laggy:
* The fps in obs source has been set to 30.
* The bit rate in OBS streaming settings has been set to 2000.
* I tried to set resolution to 1080 in source, but failed to find that option. So I set it to 1280.
* By saying 25 fps, do you mean the "common fps value" in obs streaming settings, I only find a "25fps pal" here.
Is there anything wrong with my setting? Or is there anything else I can do to improve the speed? Thank you
![image](https://github.com/ant-media/Ant-Media-Server/assets/27934653/cff5ac0b-5707-418d-b44d-0ae55c3ca322)
![image](https://github.com/ant-media/Ant-Media-Server/assets/27934653/55261a49-9a0e-40e2-8c0e-2d43eb306122)
![image](https://github.com/ant-media/Ant-Media-Server/assets/27934653/bf6211a6-5e03-4610-b4b1-8ec88414835e)
![image](https://github.com/ant-media/Ant-Media-Server/assets/27934653/ef681d4f-4592-4c50-9a4e-83e1e62448e2)

**Reply by yashtandon113** [reply]
Thanks for your cooperation, @sunpeng222 

May I know what your exact source is? Is it any local MP4 file or any browser source, etc.?

**Reply by sunpeng222** [reply]
My video source comes from my webcam, and is sent to AMS in OBS using the H.264 encoding format over RTMP.  Because I need to do motion compensation, I need high real-time performance.
![image](https://github.com/ant-media/Ant-Media-Server/assets/27934653/45971ab3-064e-4033-89ff-471f0c37ae9d)

**Reply by yashtandon113** [reply]
@sunpeng222 
So I tried to two type of webcams on my end as a source and here are my findings.

When you add a source and select the custom resolution in a list, it shows the highest resolution on top that your camera can support. For example, see the below screenshot. In this case, the highest resolution for the camera is 1280x720.

![image](https://github.com/ant-media/Ant-Media-Server/assets/82374739/e812986a-fe41-4a3a-8761-45132a0dc0bc)

So, as per the source, the below settings should also be 1280x720.

![image](https://github.com/ant-media/Ant-Media-Server/assets/82374739/71059d36-9acc-4b34-b220-7c3589fc0898)

So as I remember, you also do not see the 1080 option, which means your camera has the highest resolution of 720p, so you need to enable 720p on the server side as well.

Please check and let us know if it helps.
